htmd/parameterization/util.py

```diff
@@ -23,109 +23,49 @@ def getEquivalentsAndDihedrals(mol):
     return mol, equivalents, all_dihedrals
 
 
-def canonicalizeAtomNames(mol, fftypemethod, inplace=False, _logger=True):
+def guessElements(mol, fftypemethod):
     """
-    This fixes up the atom naming and reside name to be consistent.
-    NB this scheme matches what MATCH does.
-    Don't change it or the naming will be inconsistent with the RTF.
+    Guess element from an atom name
     """
-    if not inplace:
-        mol = mol.copy()
-    mol.segid[:] = 'L'
-    if _logger:
-        logger.info('Rename segment to %s' % mol.segid[0])
-    mol.resname[:] = 'MOL'
-    if _logger:
-        logger.info('Rename residue to %s' % mol.resname[0])
 
-    sufices = {}
-    for i in range(mol.numAtoms):
-        name = guessElementForFftype(i, mol, fftypemethod).upper()
+    elements = {}
+    elements['CGenFF_2b6'] = ['H', 'C', 'N', 'O', 'F', 'S', 'P', 'Cl', 'Br', 'I']
+    elements['GAFF']       = ['H', 'C', 'N', 'O', 'F', 'S', 'P', 'Cl', 'Br', 'I']
+    elements['GAFF2']      = ['H', 'C', 'N', 'O', 'F', 'S', 'P', 'Cl', 'Br', 'I']
 
-        sufices[name] = sufices.get(name, 0) + 1
-        name += str(sufices[name])
+    mol = mol.copy()
 
-        if _logger:
-            logger.info('Rename atom %d: %-4s --> %-4s' % (i, mol.name[i], name))
-        mol.name[i] = name
+    for i, name in enumerate(mol.name):
 
-    if not inplace:
-        return mol
+        candidates = [element for element in elements[fftypemethod] if name.capitalize().startswith(element)]
```

Comment:
What if the user passes a wrong fftypemethod? What is the reason being removing robustness to the code?